### PR TITLE
feat(typescript): new typescript projects use `projenrcTs` by default

### PR DESCRIPTION
**Pull request by mrgrain**
Sets the `projenrcTs` default value to `true` for new projects only.
We cannot change the generic default value, since migration from JS->TS requires manual effort.

To not use `projenrcTs`, pass `projen new <project-type> --projenrc-ts false`.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/new.test.ts [hunk 1]
```diff
@@ -318,7 +328,7 @@ test("projenrc-json creates node-project", () => {
 
 test("projenrc-json creates java project", () => {
   withProjectDir((projectdir) => {
-    execProjenCLI(projectdir, ["new", "java", "--projenrc-json"]);
+    execProjenCLI(projectdir, ["new", "java", "--projenrc-json", "--no-synth"]);
```
Comment (mrgrain) [inline test/new.test.ts:331]: Aligned this with other test cases. Makes the test run faster.

File: test/new.test.ts [hunk 2]
```diff
@@ -423,7 +434,13 @@ test("projen new node --outdir path/to/mydir", () => {
     shell(`mkdir -p ${join("path", "to", "mydir")}`);
 
     // WHEN
-    execProjenCLI(projectdir, ["new", "node", "--outdir", "path/to/mydir"]);
+    execProjenCLI(projectdir, [
+      "new",
+      "node",
+      "--outdir",
+      "path/to/mydir",
+      "--no-post",
```
Comment (mrgrain) [inline test/new.test.ts:442]: Aligned this with other test cases. Makes the test run faster.

**Review by Naumel — APPROVED** [review]

**Comment by samson-sham** [comment]
This breaks the extensibility of "tsconfig" & "eslintOptions" at least for my project...

**Comment by mrgrain** [comment]
Thanks for reporting this @samson-sham !

Could you please open a new issue and describe how this PR breaks extensibility of "tsconfig" & "eslintOptions" in your projects and I'm happy to look into it.

**Pull request merged**